### ISSUE TITLE
cilium: transparent encryption, kube-proxy services may use incorrect sport

### ISSUE THREAD

**Issue by jrfastab** [cilium/cilium#7122]
With encryption enabled kube-proxy services may hit a case where an icmp unreachable error happens,
```
17:04:13.410974 IP 10.10.1.1.9421 > 10.10.0.85.53: 9351+ AAAA? cilium-etcd-client.kube-system.svc.cluster.local. (66)
17:04:13.411033 IP 10.10.1.1.9422 > 10.10.0.85.53: 12894+ A? cilium-etcd-client.kube-system.svc.cluster.local. (66)
17:04:13.413912 IP 10.10.1.1 > 10.10.0.85: ICMP 10.10.1.1 udp port 9422 unreachable, length 166
17:04:13.414121 IP 10.10.1.1 > 10.10.0.85: ICMP 10.10.1.1 udp port 9421 unreachable, length 195
```
Currently iptables rules to match on encryption/decryption mark is only handled in Cilium iptables chains. However, this doesn't prevent us from hitting a kube-service iptables entry incorrectly. The fix is to move mark matching + ACCEPT action rules out of Cilium chains and it to main tables.

Even with above fix however still seeing a sport rewrite. Debugging.

**Comment by jrfastab** [comment]
https://github.com/cilium/cilium/pull/7146

**Comment by jrfastab** [comment]
Backports merged and released in 1.4.1

https://github.com/cilium/cilium/pull/7152
